Code/Engine/Core/Prefabs/Implementation/PrefabReferenceComponent.cpp

```diff
@@ -446,6 +446,33 @@ void ezPrefabReferenceComponentManager::Update(const ezWorldModule::UpdateContex
   }
 
   m_ComponentsToUpdate.Clear();
+
+  auto pWorld = GetWorld();
+
+  ezUInt32 uiGcSteps = 0;
+  auto it = GetComponents(m_uiGcStartIndex);
+
+  for (; it.IsValid() && uiGcSteps < 10; it.Next())
```

Comment:
I don't think this will work at runtime, because the prefab reference component will delete itself after it has instantiated the prefab, so you will end up with an empty game object without any components.

Reply:
I've tested it in play the game mode and there it worked, but it looks like you are right and the behavior is different in a standalone app. Have to think about that.

Reply:
Another idea would be to set a flag on the game object that this can be garbage collected when empty, and then do this collection on ALL objects.

Reply:
I was thinking about adding a component manager that does some debug vis, anyway, like rendering a marker at the location of every (empty) object, so that you can see when a prefab doesn't clean itself up.  Eg most effects at runtime are supposed to delete themselves, but might be misconfigured.